test/test_utils/sagemaker.py

```diff
@@ -0,0 +1,236 @@
+    if framework == "pytorch" and job_type == "inference":
+        # The following distutils package conflict with test dependencies
+        ec2_conn.run("apt-get remove python3-scipy python3-yaml -y")
+    ec2_conn.run(f"sudo {is_py3} pip install -r requirements.txt ", warn=True)
+    if "py3" in image and framework == "tensorflow" and job_type == "training":
+        ec2_conn.run(f"sudo {is_py3} pip install -U sagemaker-experiments")
+
+
+def execute_local_tests(image, ec2_client):
+    """
+    Run the sagemaker local tests in ec2 instance for the image
+    :param image: ECR url
+    :param ec2_client: boto3_obj
+    :return: None
+    """
+    pytest_command, path, tag, job_type = generate_sagemaker_pytest_cmd(image, SAGEMAKER_LOCAL_TEST_TYPE)
+    print(pytest_command)
+    framework = image.split("/")[1].split(":")[0].split("-")[1]
+    random.seed(f"{datetime.datetime.now().strftime('%Y%m%d%H%M%S%f')}")
+    ec2_key_name = f"{job_type}_{tag}_sagemaker_{random.randint(1, 1000)}"
+    region = os.getenv("AWS_REGION", DEFAULT_REGION)
+    sm_tests_tar_name = "sagemaker_tests.tar.gz"
+    ec2_test_report_path = os.path.join(UBUNTU_HOME_DIR, "test", f"{job_type}_{tag}_sm_local.xml")
+    try:
+        key_file = generate_ssh_keypair(ec2_client, ec2_key_name)
+        print(f"Launching new Instance for image: {image}")
+        instance_id, ip_address = launch_sagemaker_local_ec2_instance(image, UBUNTU_16_BASE_DLAMI, ec2_key_name, region)
+        ec2_conn = ec2_utils.ec2_connection(instance_id, key_file, region)
+        ec2_conn.put(sm_tests_tar_name, f"{UBUNTU_HOME_DIR}")
+        ec2_conn.run(f"$(aws ecr get-login --no-include-email --region {region})")
+        ec2_conn.run(f"docker pull {image}")
+        ec2_conn.run(f"tar -xzf {sm_tests_tar_name}")
+        with ec2_conn.cd(path):
+            install_sm_local_dependencies(framework, job_type, image, ec2_conn)
+            ec2_conn.run(pytest_command, timeout=2100)
+            print(f"Downloading Test reports for image: {image}")
+            ec2_conn.get(ec2_test_report_path, os.path.join("test", f"{job_type}_{tag}_sm_local.xml"))
+    finally:
+        print(f"Terminating Instances for image: {image}")
+        ec2_utils.terminate_instance(instance_id, region)
+        print(f"Destroying ssh Key_pair for image: {image}")
+        destroy_ssh_keypair(ec2_client, ec2_key_name)
+
+
+def install_sagemaker_remote_dependencies(framework, job_type, image, ctx):
+    if "py3" in image and framework == "tensorflow" and job_type == "training":
+        ctx.run(f"pip install -U sagemaker-experiments")
+
+
+def execute_sagemaker_remote_tests(image):
+    """
+    Run pytest in a virtual env for a particular image
+
+    Expected to run via multiprocessing
+
+    :param image: ECR url
+    """
+    pytest_command, path, tag, job_type = generate_sagemaker_pytest_cmd(image, SAGEMAKER_REMOTE_TEST_TYPE)
+    framework = image.split("/")[1].split(":")[0].split("-")[1]
```

Comment:
Please avoid using the ".split"-index method to get framework, job type, python version, processor type, etc. from the image uri.

Reply:
ok will change this regex